src/main/java/gregtech/common/covers/CoverRoboticArm.java

```diff
@@ -68,7 +68,14 @@ protected int doTransferExact(IItemHandler itemHandler, IItemHandler myItemHandl
         while (iterator.hasNext()) {
             TypeItemInfo sourceInfo = sourceItemAmount.get(iterator.next());
             int itemAmount = sourceInfo.totalCount;
+
+            // get the max we can extract from the item filter variable
             int itemToMoveAmount = itemFilterContainer.getSlotTransferLimit(sourceInfo.filterSlot);
+            int maxMultiplier = Math.floorDiv(Math.min(itemAmount, maxTransferAmount), itemToMoveAmount);
+
+            // multiply up to the total count of all the items
+            itemToMoveAmount *= Math.min(itemFilterContainer.getTransferStackSize(), maxMultiplier);
```

Comment:
I think these can be outright skipped if `maxTransferAmount` is 1, which is likely the most common case.